tool/tsh/access_request.go

```diff
@@ -289,3 +291,75 @@ func showRequestTable(reqs []types.AccessRequest) error {
 	fmt.Fprintf(os.Stderr, "\nhint: use 'tsh request show <request-id>' for additional details\n")
 	return trace.Wrap(err)
 }
+
+// rolesHaveChanged checks to see if there is a difference in the rolesets
+// applied to two certificates. Duplicates are treated as a single item and
+// the ordering of roles is ignored.
+func rolesHaveChanged(oldKey, newKey *client.Key) (bool, error) {
+	oldRoles, err := oldKey.CertRoles()
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	newRoles, err := newKey.CertRoles()
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	oldRoleSet := make(map[string]int)
+	for _, r := range oldRoles {
+		oldRoleSet[r] = 0
+	}
+
+	newRoleSet := make(map[string]int)
+	for _, r := range newRoles {
+		newRoleSet[r] = 0
+	}
```

Comment:
```suggestion
	oldRoleSet := make(map[string]struct{})
	for _, r := range oldRoles {
		oldRoleSet[r] = struct{}{}
	}

	newRoleSet := make(map[string]struct{})
	for _, r := range newRoles {
		newRoleSet[r] = struct{}{}
	}
```

Reply:
Deleted in favour of unconditionally deleting cert cache, but at least made me look up https://dave.cheney.net/2014/03/25/the-empty-struct